services/apis/indexd/indexdTasks.js

```diff
@@ -77,32 +81,85 @@ module.exports = {
     return true;  // always return true till we figure out the Promise.all issue above ...
   },
 
+  /**
+   * Fetches indexd res data for file and assigns 'rev', given an indexd file object with a did
+   * @param {Object} fileNode - Assumed to have a did property
+   * @returns {Promise<Gen3Response>}
+   */
+  async getFileFullRes(file, authHeaders = user.mainAcct.accessTokenHeader) {
+    // get data from indexd
+    return I.sendGetRequest(
+      `${indexdProps.endpoints.get}/${file.did}`,
+      authHeaders,
+    ).then((res) => {
+      file.rev = getRevFromResponse(res);
+      return res;
+    });
+  },
+
   /**
    * Fetches indexd data for file and assigns 'rev', given an indexd file object with a did
    * @param {Object} fileNode - Assumed to have a did property
    * @returns {Promise<Gen3Response>}
    */
-  async getFile(file) {
+  async getFile(file, authHeaders = user.mainAcct.accessTokenHeader) {
     // get data from indexd
     return I.sendGetRequest(
       `${indexdProps.endpoints.get}/${file.did}`,
-      user.mainAcct.accessTokenHeader,
+      authHeaders,
     ).then((res) => {
       file.rev = getRevFromResponse(res);
       return res.body;
     });
   },
 
+  /**
+   * Updates indexd data for file
+   * @param {Object} fileNode - Assumed to have a did property
+   * @returns {Promise<Gen3Response>}
```

Comment:
also update this docstring... actually lots of docstrings in this file are not up to date :(

Reply:
:sob: will update them